### ISSUE TITLE
Arm64 Support

### ISSUE THREAD

**Issue by birdinforest** [AssetRipper/AssetRipper#1154]
### Are you on the latest version of AssetRipper?

Yes, I'm on the latest release of AssetRipper.

### Which release are you using?

MacOS Arm64

### Which game did this occur on?

N/A

### Which Unity version did this occur on?

N/A

### Is the game Mono or IL2Cpp?

Unknown

### Describe the issue.

# Issue
App crashes when click on Open File and Open Folder

# System and version
Ver. [1.0.4](https://github.com/AssetRipper/AssetRipper/releases/tag/1.0.4)
System: MacOS ARM64 12.6 
Chrome: Version 118.0.5993.117 (Official Build) (arm64)

# Produce:
Compile the project. The web app display on Chrome. 
Click  Open File and Open Folder.

# Crash report:

## Crash report on web page

```
An unhandled exception occurred while processing the request.
DllNotFoundException: Unable to load shared library 'nfd' or one of its dependencies. In order to help diagnose loading problems, consider setting the DYLD_PRINT_LIBRARIES environment variable:
dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/nfd.dylib, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/nfd.dylib' (no such file)
dlopen(/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/nfd.dylib, 0x0001): tried: '/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/nfd.dylib' (no such file)
dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/nfd.dylib, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/nfd.dylib' (no such file)
dlopen(nfd.dylib, 0x0001): tried: 'nfd.dylib' (no such file), '/usr/local/lib/nfd.dylib' (no such file), '/usr/lib/nfd.dylib' (no such file), '/Users/forrrest/AssetRipper/Source/AssetRipper.GUI.Free/nfd.dylib' (no such file)
dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/libnfd.dylib, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/libnfd.dylib' (no such file)
dlopen(/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/libnfd.dylib, 0x0001): tried: '/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/libnfd.dylib' (no such file)
dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/libnfd.dylib, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/libnfd.dylib' (no such file)
dlopen(libnfd.dylib, 0x0001): tried: 'libnfd.dylib' (no such file), '/usr/local/lib/libnfd.dylib' (no such file), '/usr/lib/libnfd.dylib' (no such file), '/Users/forrrest/AssetRipper/Source/AssetRipper.GUI.Free/libnfd.dylib' (no such file)
dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/nfd, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/nfd' (no such file)
dlopen(/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/nfd, 0x0001): tried: '/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/nfd' (no such file)
dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/nfd, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/nfd' (no such file)
dlopen(nfd, 0x0001): tried: 'nfd' (no such file), '/usr/local/lib/nfd' (no such file), '/usr/lib/nfd' (no such file), '/Users/forrrest/AssetRipper/Source/AssetRipper.GUI.Free/nfd' (no such file)
dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/libnfd, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/libnfd' (no such file)
dlopen(/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/libnfd, 0x0001): tried: '/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/libnfd' (no such file)
dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/libnfd, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/libnfd' (no such file)
dlopen(libnfd, 0x0001): tried: 'libnfd' (no such file), '/usr/local/lib/libnfd' (no such file), '/usr/lib/libnfd' (no such file), '/Users/forrrest/AssetRipper/Source/AssetRipper.GUI.Free/libnfd' (no such file)
NativeFileDialogs.AutoGen.nfd+__Internal.NFD_Init()

Stack Query Cookies Headers Routing
DllNotFoundException: Unable to load shared library 'nfd' or one of its dependencies. In order to help diagnose loading problems, consider setting the DYLD_PRINT_LIBRARIES environment variable: dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/nfd.dylib, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/nfd.dylib' (no such file) dlopen(/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/nfd.dylib, 0x0001): tried: '/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/nfd.dylib' (no such file) dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/nfd.dylib, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/nfd.dylib' (no such file) dlopen(nfd.dylib, 0x0001): tried: 'nfd.dylib' (no such file), '/usr/local/lib/nfd.dylib' (no such file), '/usr/lib/nfd.dylib' (no such file), '/Users/forrrest/AssetRipper/Source/AssetRipper.GUI.Free/nfd.dylib' (no such file) dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/libnfd.dylib, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/libnfd.dylib' (no such file) dlopen(/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/libnfd.dylib, 0x0001): tried: '/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/libnfd.dylib' (no such file) dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/libnfd.dylib, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/libnfd.dylib' (no such file) dlopen(libnfd.dylib, 0x0001): tried: 'libnfd.dylib' (no such file), '/usr/local/lib/libnfd.dylib' (no such file), '/usr/lib/libnfd.dylib' (no such file), '/Users/forrrest/AssetRipper/Source/AssetRipper.GUI.Free/libnfd.dylib' (no such file) dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/nfd, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/nfd' (no such file) dlopen(/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/nfd, 0x0001): tried: '/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/nfd' (no such file) dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/nfd, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/nfd' (no such file) dlopen(nfd, 0x0001): tried: 'nfd' (no such file), '/usr/local/lib/nfd' (no such file), '/usr/lib/nfd' (no such file), '/Users/forrrest/AssetRipper/Source/AssetRipper.GUI.Free/nfd' (no such file) dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/libnfd, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/libnfd' (no such file) dlopen(/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/libnfd, 0x0001): tried: '/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/libnfd' (no such file) dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/libnfd, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/libnfd' (no such file) dlopen(libnfd, 0x0001): tried: 'libnfd' (no such file), '/usr/local/lib/libnfd' (no such file), '/usr/lib/libnfd' (no such file), '/Users/forrrest/AssetRipper/Source/AssetRipper.GUI.Free/libnfd' (no such file)
NativeFileDialogs.AutoGen.nfd+__Internal.NFD_Init()
NativeFileDialogs.AutoGen.nfd.NFD_Init()
NativeFileDialogs.Net.NfdManager.PushDialog()
NativeFileDialogs.Net.Nfd.OpenDialogMultiple(out string[] outPaths, IDictionary<string, string> filters, string defaultPath)
AssetRipper.GUI.Web.Dialogs+OpenFiles.GetUserInput(out string[] paths, IDictionary<string, string> filters, string defaultPath) in Dialogs.cs
-
		}
		public static NfdStatus GetUserInput(out string[]? paths, IDictionary<string, string>? filters = null, string? defaultPath = null)
		{
			lock (lockObject)
			{
				return Nfd.OpenDialogMultiple(out paths, filters, defaultPath);
			}
		}
	}
	public static class OpenFile
	{
AssetRipper.GUI.Web.Pages.Commands+LoadFile.AssetRipper.GUI.Web.Pages.ICommand.Start(HttpRequest request) in Commands.cs
-
			if (form.TryGetValue("Path", out StringValues values))
			{
				paths = values;
			}
			else
			{
				Dialogs.OpenFiles.GetUserInput(out paths);
			}
			if (paths is { Length: > 0 })
			{
				GameFileLoader.LoadAndProcess(paths);
			}
Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl.Invoke(HttpContext context)

Show raw exception details
System.DllNotFoundException: Unable to load shared library 'nfd' or one of its dependencies. In order to help diagnose loading problems, consider setting the DYLD_PRINT_LIBRARIES environment variable: 
dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/nfd.dylib, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/nfd.dylib' (no such file)
dlopen(/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/nfd.dylib, 0x0001): tried: '/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/nfd.dylib' (no such file)
dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/nfd.dylib, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/nfd.dylib' (no such file)
dlopen(nfd.dylib, 0x0001): tried: 'nfd.dylib' (no such file), '/usr/local/lib/nfd.dylib' (no such file), '/usr/lib/nfd.dylib' (no such file), '/Users/forrrest/AssetRipper/Source/AssetRipper.GUI.Free/nfd.dylib' (no such file)
dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/libnfd.dylib, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/libnfd.dylib' (no such file)
dlopen(/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/libnfd.dylib, 0x0001): tried: '/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/libnfd.dylib' (no such file)
dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/libnfd.dylib, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/libnfd.dylib' (no such file)
dlopen(libnfd.dylib, 0x0001): tried: 'libnfd.dylib' (no such file), '/usr/local/lib/libnfd.dylib' (no such file), '/usr/lib/libnfd.dylib' (no such file), '/Users/forrrest/AssetRipper/Source/AssetRipper.GUI.Free/libnfd.dylib' (no such file)
dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/nfd, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/nfd' (no such file)
dlopen(/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/nfd, 0x0001): tried: '/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/nfd' (no such file)
dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/nfd, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/nfd' (no such file)
dlopen(nfd, 0x0001): tried: 'nfd' (no such file), '/usr/local/lib/nfd' (no such file), '/usr/lib/nfd' (no such file), '/Users/forrrest/AssetRipper/Source/AssetRipper.GUI.Free/nfd' (no such file)
dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/libnfd, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/libnfd' (no such file)
dlopen(/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/libnfd, 0x0001): tried: '/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/libnfd' (no such file)
dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/libnfd, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/libnfd' (no such file)
dlopen(libnfd, 0x0001): tried: 'libnfd' (no such file), '/usr/local/lib/libnfd' (no such file), '/usr/lib/libnfd' (no such file), '/Users/forrrest/AssetRipper/Source/AssetRipper.GUI.Free/libnfd' (no such file)

   at NativeFileDialogs.AutoGen.nfd.__Internal.NFD_Init()
   at NativeFileDialogs.AutoGen.nfd.NFD_Init()
   at NativeFileDialogs.Net.NfdManager.PushDialog()
   at NativeFileDialogs.Net.Nfd.OpenDialogMultiple(String[]& outPaths, IDictionary`2 filters, String defaultPath)
   at AssetRipper.GUI.Web.Dialogs.OpenFiles.GetUserInput(String[]& paths, IDictionary`2 filters, String defaultPath) in /Users/forrrest/AssetRipper/Source/AssetRipper.GUI.Web/Dialogs.cs:line 25
   at AssetRipper.GUI.Web.Pages.Commands.LoadFile.AssetRipper.GUI.Web.Pages.ICommand.Start(HttpRequest request) in /Users/forrrest/AssetRipper/Source/AssetRipper.GUI.Web/Pages/Commands.cs:line 21
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl.Invoke(HttpContext context)
```

# Log on console
```
fail: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware[1]
      An unhandled exception has occurred while executing the request.
      System.DllNotFoundException: Unable to load shared library 'nfd' or one of its dependencies. In order to help diagnose loading problems, consider setting the DYLD_PRINT_LIBRARIES environment variable: 
      dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/nfd.dylib, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/nfd.dylib' (no such file)
      dlopen(/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/nfd.dylib, 0x0001): tried: '/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/nfd.dylib' (no such file)
      dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/nfd.dylib, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/nfd.dylib' (no such file)
      dlopen(nfd.dylib, 0x0001): tried: 'nfd.dylib' (no such file), '/usr/local/lib/nfd.dylib' (no such file), '/usr/lib/nfd.dylib' (no such file), '/Users/forrrest/AssetRipper/Source/AssetRipper.GUI.Free/nfd.dylib' (no such file)
      dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/libnfd.dylib, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/libnfd.dylib' (no such file)
      dlopen(/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/libnfd.dylib, 0x0001): tried: '/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/libnfd.dylib' (no such file)
      dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/libnfd.dylib, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/libnfd.dylib' (no such file)
      dlopen(libnfd.dylib, 0x0001): tried: 'libnfd.dylib' (no such file), '/usr/local/lib/libnfd.dylib' (no such file), '/usr/lib/libnfd.dylib' (no such file), '/Users/forrrest/AssetRipper/Source/AssetRipper.GUI.Free/libnfd.dylib' (no such file)
      dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/nfd, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/nfd' (no such file)
      dlopen(/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/nfd, 0x0001): tried: '/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/nfd' (no such file)
      dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/nfd, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/nfd' (no such file)
      dlopen(nfd, 0x0001): tried: 'nfd' (no such file), '/usr/local/lib/nfd' (no such file), '/usr/lib/nfd' (no such file), '/Users/forrrest/AssetRipper/Source/AssetRipper.GUI.Free/nfd' (no such file)
      dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/libnfd, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/runtimes/osx-arm64/native/libnfd' (no such file)
      dlopen(/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/libnfd, 0x0001): tried: '/usr/local/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/libnfd' (no such file)
      dlopen(/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/libnfd, 0x0001): tried: '/Users/forrrest/AssetRipper/Source/0Bins/AssetRipper.GUI.Free/Debug/libnfd' (no such file)
      dlopen(libnfd, 0x0001): tried: 'libnfd' (no such file), '/usr/local/lib/libnfd' (no such file), '/usr/lib/libnfd' (no such file), '/Users/forrrest/AssetRipper/Source/AssetRipper.GUI.Free/libnfd' (no such file)
      
         at NativeFileDialogs.AutoGen.nfd.__Internal.NFD_Init()
         at NativeFileDialogs.AutoGen.nfd.NFD_Init()
         at NativeFileDialogs.Net.NfdManager.PushDialog()
         at NativeFileDialogs.Net.Nfd.OpenDialogMultiple(String[]& outPaths, IDictionary`2 filters, String defaultPath)
         at AssetRipper.GUI.Web.Dialogs.OpenFiles.GetUserInput(String[]& paths, IDictionary`2 filters, String defaultPath) in /Users/forrrest/AssetRipper/Source/AssetRipper.GUI.Web/Dialogs.cs:line 25
         at AssetRipper.GUI.Web.Pages.Commands.LoadFile.AssetRipper.GUI.Web.Pages.ICommand.Start(HttpRequest request) in /Users/forrrest/AssetRipper/Source/AssetRipper.GUI.Web/Pages/Commands.cs:line 21
         at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl.Invoke(HttpContext context)

```

# others
Tested released build 1.0.4. It crashes as well with error:
```
[1]    66552 segmentation fault  ./AssetRipper.GUI.Free
```

### Relevant log output

[AssetRipper.log](https://github.com/AssetRipper/AssetRipper/files/13874499/AssetRipper.log)

**Comment by birdinforest** [comment]
Thank you, @ds5678. Could you provide me with some guidance on addressing Arm64 support? I am interested in contributing to it, but I lack knowledge about the system compatibility of .NET. Thanks.

**Comment by ds5678** [comment]
AssetRipper does not use the repository you linked. If you would like to work on Arm64 support, a good place would be:

https://github.com/AssetRipper/nativefiledialog-extended-builds

**Comment by birdinforest** [comment]
I built nativefiledialog-extended target to arm64 and linked it to AssetRipper. The file dialog UI leads to error compaing the UI is processed outside main thread: `"NSWindow drag regions should only be invalidated on the Main Thread!"`.

I don't think that issue caused by NFD, as according to a [comment](https://github.com/btzy/nativefiledialog-extended/issues/125#issuecomment-1879767237) from developer:
> NFD isn't really intended to be invoked from multiple threads, and AFAIK MacOS at least doesn't like it if you do UI-related things from another thread. 

I tried to refactor the file `Dialogs.cs` to run file dialog UI with async, it throws out the same error. According to my understanding the web page is a client of a AspNetCore app, there is no `mulitple threads` processing on client. I am not sure if it right or not.

Any advice about it?

Btw, file import, unpack, and export works well via Commands on my mac arm64, so that it looks like the only issue is the file dialog. Is it an option to create a web UI file dialog to avoid system compatibility?

**Comment by ds5678** [comment]
> file import, unpack, and export works well via Commands on my mac arm64

If you're self-compiling, that's why.

> there is no mulitple threads processing on client

There is no processing at all on the client (web page). It happens in the server, which is multithreaded.

> Any advice?

I am exhausted at the moment, but I'll read over your issue again when I wake up.

**Comment by birdinforest** [comment]
> There is no processing at all on the client (web page). It happens in the server, which is multithreaded.

Yes, now I get it that webpages are generated on server. The file dialog UI is starts on server as well.

**Comment by ds5678** [comment]
> I think it is a better cross-platform solution to generate HTML Input for file input and export.

I tried that. The browser only gives file names, rather than full paths.

**Comment by birdinforest** [comment]
> > I think it is a better cross-platform solution to generate HTML Input for file input and export.
> 
> I tried that. The browser only gives file names, rather than full paths.

Yes I didn't realise that there is a strict security limit.

On Mac ARM64, Nfd doesn't work in the Task process, however, it works well alone in a separate process. I tried setting up a named pipe server to run Nfd and transmit the file path to AssetStripper. It works as expected.

Do you think it is a reasonable solution, only on Mac?

**Comment by ds5678** [comment]
I would have to see what that looks like before making a judgement.

**Comment by cihadturhan** [comment]
> Arm64 isn't supported at the moment. I'm actually surprised you were able to get it to run at all.

Why isn't this run with rosetta? It should emulate intel based apps without issue or am I wrong?

**Comment by wwwDayDream** [comment]
Bump as I'm having this issue on Windows 10 :(

**Comment by ds5678** [comment]
@wwwDayDream I don't understand how that's related to Arm64 support.

**Comment by ds5678** [comment]
Implemented in 90577d1409d4b340114be5647e2252d3dde1a880
